toxcore/network.c

```diff
@@ -29,6 +31,12 @@
 #include "config.h"
 #endif
 
+#ifdef __APPLE__
+#define _DARWIN_C_SOURCE
+#include <mach/clock.h>
+#include <mach/mach.h>
+#endif
```

Comment:
I assume you moved this block so you don't need an extra block at the beginning. Would it work if you moved this block back to where it came from and `#define _DARWIN_C_SOURCE` unconditionally together with `_XOPEN_SOURCE`?